src/utils.py

```diff
@@ -0,0 +1,29 @@
+import os
+import logging
+
+def delete_empty_folders(root):
+    deleted = set()
+    for current_dir, subdirs, files in os.walk(root, topdown=False):
+        still_has_subdirs = False
+        for subdir in subdirs:
+            if os.path.join(current_dir, subdir) not in deleted:
+                still_has_subdirs = True
+                break
+        if not any(files) and not still_has_subdirs:
+            os.rmdir(current_dir)
+            logging.info("Deleted empty folder: %s", current_dir)
+            deleted.add(current_dir)
+    return deleted
+
+def rename_rendered_files(job_rendered_path):
+    """walk the rendered path and mv the files to the correct location
+
+    Args:
+        job_rendered_path (str): _description_
+    """
+    for root, _dirs, files in os.walk(job_rendered_path):
```

Comment:
since we already know the path in advance, won't it be easier to just move everything under the path?

Reply:
We only know the root in advance, we dont know how it will be rendered inside it, its a github action so its has to be "stupid" to work on this path.
meaning, i can have a 
`job_rendered_path = /devops-jobs` and underneath it
`rendered/devops-jobs/db-user-manger/dev.yaml`
`rendered/devops-jobs/ecr-manager/dev.yaml`

and i can also have 
`rendered/backend-api/dev.yaml`

i didnt checked my code yet on the "microservice" part, but the idea is we know the root path but we dont know how the tree underneath it looks like